rewrite-gradle/src/main/java/org/openrewrite/gradle/GradleParser.java

```diff
@@ -44,8 +44,10 @@ public class GradleParser implements Parser {
     @Override
     public Stream<SourceFile> parseInputs(Iterable<Input> sources, @Nullable Path relativeTo, ExecutionContext ctx) {
         if (buildParser == null) {
-            if (base.buildscriptClasspath == null && defaultClasspath == null) {
-                defaultClasspath = loadDefaultClasspath();
+            if (base.buildscriptClasspath == null) {
+                if (defaultClasspath == null) {
+                    defaultClasspath = loadDefaultClasspath();
+                }
                 base.buildscriptClasspath = defaultClasspath;
```

Comment:
It may be safer to instead use a local variable that way if the end user has a handle to the base builder, then we aren't modifying the state of that instance from here as a side effect.

Reply:
Good point 👍